internal/groups/key.go

```diff
@@ -69,5 +79,48 @@ func (g *GroupKeyFromMetadata) GetGroupKey(node *v1.Node) GroupKey {
 	// let's build the groups values from labels and annotations
 	values = append(getValueOrEmpty(node.Labels, g.labelsKeys), getValueOrEmpty(node.Annotations, g.annotationKeys)...)
 
+	if podOverride, hasPodOverride := g.getGroupKeyFromPods(node); hasPodOverride {
+		return podOverride
+	}
```

Comment:
I would keep the group key generation in this method
```suggestion
	if podOverride, hasPodOverride := g.getGroupKeyFromPods(node); hasPodOverride {
		values = podOverride
	}
```

Reply:
done

Reply:
well `values = []string{string(podOverride)}` in fact